pkg/restmapper/caching.go

```diff
@@ -70,12 +72,35 @@ type cachedGVR struct {
 	Scope    meta.RESTScope
 }
 
+// KindFromGVR finds out the Kind from the GVR in the cache. If the GVR version is not given, we will iterate all the matching
+// GR in the cache and return the first matching one.
+// e.g. https://github.com/kubernetes/kubernetes/blob/a0cff30104ea950a5cc733a109e7f9084275e49e/staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset.go#L353
+func (c *cache) KindFromGVR(gvr schema.GroupVersionResource) string {
+	if gvr.Version != "" {
+		cachedgvr, ok := c.groupVersions[gvr.GroupVersion()]
+		if !ok {
+			return ""
+		}
+		return cachedgvr.toKind[gvr.Resource]
+	}
+	for keyGVR, cachedgvr := range c.groupVersions {
+		if keyGVR.Group != gvr.Group {
```

Comment:
Do we really do this in KindFor?  It seems that would often return an error, given the docs for KindFor say "KindFor takes a partial resource and returns the single match.  Returns an error if there are multiple matches"

Reply:
reverted.